velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -401,6 +401,11 @@ RowVectorPtr NestedLoopJoinProbe::doMatch(vector_size_t probeCnt) {
   VELOX_CHECK(!hasProbedAllBuildData());
 
   if (joinCondition_ == nullptr) {
+    // All rows in SelectivityVector for probe and build must be marked valid
+    // for a CrossJoin.
+    probeMatched_.setAll();
```

Comment:
It seems inefficient to reset unused SelectivityVectors. Should we look into modifying the needsProbeMismatch and needsBuildMismatch instead?

Reply:
Agreed. An approach to support "left/right/full cross join" would be returning false from `needsProbe/BuildMisatch` when `joinCondition_ == nullptr && joinType_ != innerJoin`.

Reply:
A third option could be to change buildMismatchedOutput to return nullptr if joinCondition_ = nullptr
https://github.com/facebookincubator/velox/blob/main/velox/exec/NestedLoopJoinProbe.cpp#L225

It seemed needsProbe/BuildMismatch is well abstracted to just look at joinType to determine if mismatching is needed.